packages/elements/src/datetime-field/index.ts

```diff
@@ -0,0 +1,708 @@
+    return isNaN(value.getTime()) ? '' : utcFormat(value, this.locale.isoFormat);
+  }
+
+  /**
+   * Returns true if the datetime field has timepicker
+   * @returns hasTimePicker
+   */
+  protected get hasTimePicker (): boolean {
+    // need to check for attribute to resolve the value correctly until the first lifecycle is run
+    return this.timepicker || this.hasAttribute('timepicker') || this.hasAmPm || this.hasSeconds;
+  }
+
+  /**
+   * Returns true if the datetime field has seconds
+   * @returns hasSeconds
+   */
+  protected get hasSeconds (): boolean {
+    return this.showSeconds || this.hasAttribute('show-seconds');
+  }
+
+  /**
+   * Returns true if the datetime field has am-pm
+   * @returns hasAmPm
+   */
+  protected get hasAmPm (): boolean {
+    return this.amPm || this.hasAttribute('am-pm');
+  }
+
+  /**
+   * Get the start date for when the value is not set
+   */
+  protected get startDate (): string {
+    // Noon in UTC
+    const today = new Date();
+    const date = new Date(0);
+    date.setUTCFullYear(today.getFullYear());
+    date.setUTCMonth(today.getMonth());
+    date.setUTCDate(today.getDate());
+    date.setUTCHours(12);
+    return this.dateToString(date);
+  }
+
+  /**
+   * Updates the element
+   * @param changedProperties Properties that has changed
+   * @returns {void}
+   */
+  public willUpdate (changedProperties: PropertyValues): void {
+    super.willUpdate(changedProperties);
+
+    if (changedProperties.has(TranslatePropertyKey)) {
+      this._locale = null; // Locale is updated on next call via getter
+    }
+
+    if (changedProperties.has(FocusedPropertyKey) && !this.focused) {
+      this.partLabel = '';
+    }
+  }
+
+  /**
+   * Check if input value should be synchronised with component value
+   * @param changedProperties Properties that has changed
+   * @returns True if input should be synchronised
+   */
+  protected override shouldSyncInputValue (changedProperties: PropertyValues): boolean {
+    // Note: changing any of these properties override the input value
+    // On blur, if the value is correct makes ure strict format is used
+    return changedProperties.has('interimValueState')
+      || changedProperties.has(TranslatePropertyKey)
+      || changedProperties.has('formatOptions')
+      || changedProperties.has('timepicker')
+      || changedProperties.has('showSeconds')
+      || changedProperties.has('amPm')
+      || (changedProperties.has(FocusedPropertyKey) && this.value !== '' && !this.focused);
+  }
+
+  /**
+   * Synchronise input value with value.
+   * Override the method if value and input value are not the same
+   * @param changedProperties Properties that has changed
+   * @returns {void}
+   */
+  // eslint-disable-next-line @typescript-eslint/no-unused-vars
+  protected override syncInputValue (changedProperties: PropertyValues): void;
+  protected syncInputValue (): void;
+  protected syncInputValue (changedProperties?: PropertyValues): void {
```

Comment:
My eslint warning only go away when the lines are like this. 
```suggestion
  // eslint-disable-next-line @typescript-eslint/no-unused-vars
  protected syncInputValue (changedProperties?: PropertyValues): void {
```

Reply:
I also curious why there are multiple method declarations with the same name, why can't we just override it only once, can you elaborate ?

Reply:
There are two name declarations, one from super class (with changed properties) and another one without. But I agree this can be done better.

Reply:
I moved the comment to a different line, but otherwise I think it is fine as it is, as the purpose of the function is to syncInputValue :)